h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -995,7 +991,7 @@ private void setWriteVersion(long version) {
      *
      * @return the new version
      */
-    public long commit() {
+    public synchronized long commit() {
```

Comment:
Strictly speaking, there is a race condition (or even possibility of corruption) in 
`long v = ++currentVersion;`